components/renku_data_services/project/db.py

```diff
@@ -656,3 +665,150 @@ async def delete_session_secret_slot(
                 )
 
             await session.delete(secret_slot)
+
+    async def get_all_session_secrets_from_project(
+        self,
+        user: base_models.APIUser,
+        project_id: ULID,
+    ) -> list[models.SessionSecret]:
+        """Get all session secrets from a project."""
+        if user.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+
+        # Check that the user is allowed to access the project
+        authorized = await self.authz.has_permission(user, ResourceType.project, project_id, Scope.READ)
+        if not authorized:
+            raise errors.MissingResourceError(
+                message=f"Project with id '{project_id}' does not exist or you do not have access to it."
+            )
+
+        async with self.session_maker() as session:
+            result = await session.scalars(
+                select(schemas.SessionSecretORM)
+                .where(schemas.SessionSecretORM.user_id == user.id)
+                .where(schemas.SessionSecretORM.secret_slot_id == schemas.SessionSecretSlotORM.id)
+                .where(schemas.SessionSecretSlotORM.project_id == project_id)
+            )
+            secrets = result.all()
+
+            return [s.dump() for s in secrets]
+
+    async def patch_session_secrets(
+        self,
+        user: base_models.APIUser,
+        project_id: ULID,
+        secrets: list[models.SessionSecretPatchExistingSecret | models.SessionSecretPatchSecretValue],
+    ) -> list[models.SessionSecret]:
+        """Create, update or remove session secrets."""
+        if user.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+
+        # Check that the user is allowed to access the session launcher
+        authorized = await self.authz.has_permission(user, ResourceType.project, project_id, Scope.READ)
+        if not authorized:
+            raise errors.MissingResourceError(
+                message=f"Project with id '{project_id}' does not exist or you do not have access to it."
+            )
+
```

Comment:
I think this would be nicer to address as a cooldown task, i.e. maybe have a local subclass for common errors, e.g. `MissingProjectById`, `MissingProjectBySlug`.